### PR TITLE
#6 Task. Gaziz_Tuleukulov

### PR DESCRIPTION
**Pull request by Gaziz666**
https://travis-ci.org/github/Gaziz666/js-assignments/builds/707337210

### CODE REVIEW

**Comment by elizavetaShcherbatykhAis** [comment]
Your PR should contain files related only to this task, here I see the files of previous tasks. They shouldn't be here. Please remove these files from the PR.
I'll mark the PR as draft please click "ready for review" when it will be finished!
Thank you!

**Comment by Gaziz666** [comment]
I fixed the mistakes, and only 2 fails changed
.travis.yml and 6 task

File: task/06-conditions-n-loops-tasks.js [hunk 1]
```diff
@@ -82,7 +97,10 @@ function getSumBetweenNumbers(n1, n2) {
  *   10,10,10 =>  true
  */
 function isTriangle(a,b,c) {
-    throw new Error('Not implemented');
+    if ( a + b > c && a + c > b && b + c > a){
+        return true;
+    } 
+    return false
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js]: You can return it in 1 row, like:
return  a + b > c && a + c > b && b + c > a;
Expression will be truthfully or falsy :)

File: task/06-conditions-n-loops-tasks.js [hunk 2]
```diff
@@ -119,7 +137,10 @@ function isTriangle(a,b,c) {
  *  
  */
 function doRectanglesOverlap(rect1, rect2) {
-    throw new Error('Not implemented');
+    if(rect1.top + rect1.height > rect2.top && rect1.left + rect1.width > rect2.left) {
+        return true
+    } 
+    return false
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js]: Same

File: task/06-conditions-n-loops-tasks.js [hunk 3]
```diff
@@ -150,7 +171,13 @@ function doRectanglesOverlap(rect1, rect2) {
...
  */
 function isInsideCircle(circle, point) {
-    throw new Error('Not implemented');
+    let x = point.x - circle.center.x;
+    let y = point.y - circle.center.y;
+    let line = Math.sqrt(x*x + y*y);
+    if(line < circle.radius) {
+        return true
+    } 
+    return false
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js]: Same

File: task/06-conditions-n-loops-tasks.js [hunk 4]
```diff
@@ -166,7 +193,21 @@ function isInsideCircle(circle, point) {
...
+    for (let i = 0; i < arr.length -1; i++) {
+        if(arr[i] !== arr[i-1] && arr[i] !== arr[i+1]){
+            return arr[i];
+        }
+    }
+    if(arr[arr.length - 2] !== arr[arr.length - 1]) {
+        return arr[arr.length -1]
+    } 
+    return null
+
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js]: You can do it easier and shorter ;) Please try to improve your code. Thank you!

File: task/06-conditions-n-loops-tasks.js [hunk 5]
```diff
@@ -226,7 +279,8 @@ function reverseString(str) {
  *   34143 => 34143
  */
 function reverseInteger(num) {
-    throw new Error('Not implemented');
+    let str = new String(num)
+    return +str.split('').reverse().join('')
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js]: You can use toString() method and return all in 1 row

File: task/06-conditions-n-loops-tasks.js [hunk 6]
```diff
@@ -296,7 +369,32 @@ function getDigitalRoot(num) {
...
+            } else if(stack[stack.length - 1] !== closeIndex) {
+                return false
+            }
+            stack.pop()          
+        }
+    }
+    if (stack.length === 0) {
+        return true
+    } 
+    return false
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js]: You can return it like:
return stack.length === 0

File: task/06-conditions-n-loops-tasks.js [hunk 7]
```diff
@@ -270,7 +343,7 @@ function isCreditCardNumber(ccn) {
  *   165536 (1+6+5+5+3+6 = 26,  2+6 = 8) => 8
  */
 function getDigitalRoot(num) {
-    throw new Error('Not implemented');
+    return num % 9
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js]: Try to solve this function in a different way, based on the description. Thank you!

File: task/06-conditions-n-loops-tasks.js [hunk 8]
```diff
@@ -371,9 +501,34 @@ function toNaryString(num, n) {
  *   ['/web/assets/style.css', '/web/scripts/app.js',  'home/setting.conf'] => ''
  *   ['/web/assets/style.css', '/.bin/mocha',  '/read.me'] => '/'
  *   ['/web/favicon.ico', '/web-scripts/dump', '/webalizer/logs'] => '/'
+ * [ "", "web", "images", "image1.png"]
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js]: Please, don't change the description

**Comment by elizavetaShcherbatykhAis** [comment]
Super! Please check the other comments too.
I'll mark the PR as draft please click "ready for review" when it will be finished!
Thank you!

**Comment by Gaziz666** [comment]
I hope it's better

**Comment by elizavetaShcherbatykhAis** [comment]
Yes, super! I close PR as accepted

**Pull request closed without merging**